lxd/apparmor/archive.go

```diff
@@ -0,0 +1,120 @@
+package apparmor
+
+import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+	"text/template"
+
+	"github.com/lxc/lxd/lxd/sys"
+	"github.com/lxc/lxd/shared"
+)
+
+var archiveProfileTpl = template.Must(template.New("archiveProfile").Parse(`#include <tunables/global>
+profile "{{.name}}" {
+  #include <abstractions/base>
+  #include <abstractions/nameservice>
+
+{{range $index, $element := .allowedCommandsPath}}
+  {{$element}} mixr,
+{{- end }}
+
+  {{ .outputPath }}/ rw,
+  {{ .outputPath }}/** rw,
+  {{ .varPath }}/backups/** rw,
+  {{ .varPath }}/images/** r,
+
+  signal (receive) set=("term") peer=unconfined,
+
+  # Capabilities
+  capability dac_read_search,
+  capability dac_override,
+  capability chown,
+  capability fsetid,
+
+{{- if .snap }}
+  # Snap-specific libraries
+  /snap/lxd/*/lib/**.so*                  mr,
+{{- end }}
+}
+`))
+
+// ArchiveLoad ensures that the archive's policy is loaded into the kernel.
+func ArchiveLoad(sysOS *sys.OS, outputPath string, allowedCommandsPath []string) error {
```

Comment:
For clarity please can you rename `allowedCommandsPath` to `allowedCommandPaths`, as otherwise having the "s" on "commands" rather than "paths" suggests that only a single path is allowed that will be used for all of the allowed commands to be run from (which then conflicts with the fact its a string slice argument).

There's a few other places where this needs updating too.

Reply:
Fixed.